### PR TITLE
Warn on unknown fields in library/executable/test sections

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: src/Config.hs [hunk 1]
```diff
@@ -25,10 +27,20 @@ import qualified Data.Text as T
...
-data CaptureUnknownFields a = CaptureUnknownFields [String] a
+genericParseJSON_ :: forall a. (Typeable a, Generic a, GFromJSON (Rep a)) => Value -> Parser a
+genericParseJSON_ = genericParseJSON defaultOptions {fieldLabelModifier = hyphenize name}
+  where
+    name = (tyConName . typeRepTyCon . typeRep) (Proxy :: Proxy a)
+
+hyphenize :: String -> String -> String
+hyphenize name = camelTo '-' . drop (length name)
+
+data CaptureUnknownFields a = CaptureUnknownFields {captureUnknownFieldsFields :: [String], captureUnknownFieldsValue :: a}
```
Comment (soenkehahn) [inline src/Config.hs]: Could we break up this line?

File: src/Config.hs [hunk 2]
```diff
@@ -170,9 +182,9 @@ mkPackage (CaptureUnknownFields unknownFields PackageConfig{..}) = do
   let sourceDirs = fromMaybeList packageConfigSourceDirs
   let defaultExtensions = fromMaybeList packageConfigDefaultExtensions
   let ghcOptions = fromMaybeList packageConfigGhcOptions
-  mLibrary <- mapM (mkLibrary sourceDirs dependencies defaultExtensions ghcOptions) packageConfigLibrary
-  executables <- toExecutables sourceDirs dependencies defaultExtensions ghcOptions packageConfigExecutables
-  tests <- toExecutables sourceDirs dependencies defaultExtensions ghcOptions packageConfigTests
+  mLibrary <- mapM (mkLibrary sourceDirs dependencies defaultExtensions ghcOptions) (captureUnknownFieldsValue <$> packageConfigLibrary)
+  executables <- toExecutables sourceDirs dependencies defaultExtensions ghcOptions (fmap captureUnknownFieldsValue <$> packageConfigExecutables)
+  tests <- toExecutables sourceDirs dependencies defaultExtensions ghcOptions (fmap captureUnknownFieldsValue <$> packageConfigTests)
```
Comment (soenkehahn) [inline src/Config.hs]: Maybe break up these lines too?

**Comment by soenkehahn** [comment]
Apart from comments LGTM.

File: src/Config.hs (hunk 1 above)
Comment (sol) [inline src/Config.hs]: Yes, good point.

File: src/Config.hs (hunk 2 above)
Comment (sol) [inline src/Config.hs]: Done in 926e3f3f9860715a730dd1086e4ff4b7916dcce6

**Pull request merged**